core/backend/src/rpc-impl/RpcBriefcaseUtility.ts

```diff
@@ -95,6 +95,15 @@ export class RpcBriefcaseUtility {
     }
   }
 
+  public static async findOrOpen(requestContext: AuthorizedClientRequestContext, iModel: IModelRpcProps, syncMode: SyncMode): Promise<IModelDb> {
+    const iModelDb = IModelDb.tryFindByKey(iModel.key);
+    if (undefined === iModelDb) {
+      return this.open({ requestContext, tokenProps: iModel, syncMode, timeout: 1000 });
+    }
+    await iModelDb.reattachDaemon(requestContext);

```

Comment:
well, since the only thing after the await is `return iModel`, this won't change anything, but if we're following the rules strictly, yes, there should be a call to enter here.

Reply:
Well really my concern is with all the awaits in RPC method implementations where you call this - if we just add the extra enter here, we shouldn't need to do it after every call to `findOrOpen` in `IModelReadRpcImpl`, `TileRequestMemoizer`, etc.

Without changing one of the two (here or each call site), I think the ActivityID can get out-of-sync...

Reply:
Ah, right. I forgot to add it to the RPC implementation classes. But if I understand the event loop correctly, just adding it here wouldn't solve it: the scheduler can queue microtasks after the return statement here but before continuing the method that called this one. That could throw off the activity id too.

Reply:
just a comment on the whole `requestContext.enter()` nonsense. Having a global variable that holds the "current context" is a complete disaster and should be removed. It is only there for logging, and our whole strategy for logging is backwards. We should log when we catch errors, *not* when we throw them, and then we should always pass whatever "context" is necessary from the arguments to the failing function. Auto-logging on throws is wrong, and having "magic context" for logging is even worse.

Sorry @paulius-valiunas this has nothing to do with this PR, but all the requestContext.enter crap is annoying as hell to me.

Reply:
@kabentley I agree on the pain points of `requestContext.enter()` and it's something I'd like to visit for 3.0 as it has implication on decoupling things as well.

Reply:
@wgoehrig I added `requestContext.enter()` to every method that uses `findOrOpen`, we should be good now